src/main/java/Entities/Group.java

```diff
@@ -0,0 +1,89 @@
+package main.java.Entities;
+import main.java.Entities.User;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class Group {
+    /** Creates a Group object */
+    private final String groupName;
+    private String groupCode;
+    private final ArrayList<String> groupMembers;
+    private final ArrayList<String> usedCodes;
+    private final RecipeBook recipeBook;
+
+
+    /**
+     * Creates a group with a name, group code, group members, and Recipe Book.
+     * @param groupName - the name of the group
+     */
+    public Group(String groupName) {
+        this.groupName = groupName;
+        this.groupCode = "";
+        this.groupMembers = new ArrayList<>();
+        this.recipeBook = new RecipeBook();
+        this.usedCodes = new ArrayList<>();
+    }
+
+
+    /**
+     * Generates a unique 7 digit group code and assign the code to the group.
+     * @param group - the name of the group
+     */
+    public String generateGroupCode(Group group) {
```

Comment:
This definitely violates the single responsibility principle. Lets move this to a GroupFactory whose job is to create a group properly? That way, we maintain a single responsibility in the Group Entity, use a design pattern appropriately, and the problem on line 23 would be taken care of as every group initialized anywhere in the program would be created safely.